source/fuzz/transformation_add_dead_break.cpp

```diff
@@ -145,22 +145,10 @@ bool TransformationAddDeadBreak::IsApplicable(
     return false;
   }
 
-  assert(bb_from != nullptr &&
-         "We should have found a block if this line of code is reached.");
-  assert(
-      bb_from->id() == message_.from_block() &&
-      "The id of the block we found should match the source id for the break.");
-  assert(bb_to != nullptr &&
-         "We should have found a block if this line of code is reached.");
-  assert(
-      bb_to->id() == message_.to_block() &&
-      "The id of the block we found should match the target id for the break.");
-
   // Check whether the data passed to extend OpPhi instructions is appropriate.
-  if (!fuzzerutil::PhiIdsOkForNewEdge(context, bb_from, bb_to,
-                                      message_.phi_id())) {
-    return false;
-  }
+  assert(fuzzerutil::PhiIdsOkForNewEdge(context, bb_from, bb_to,
+                                        message_.phi_id()) &&
+         "Provided ids for OpPhi instruction are invalid");
```

Comment:
We should keep this as a conditional.

The thing is that while the fuzzer is in control of which transformations get created, and will never create a transformation with invalid phi ids, the *shrinker* randomly removes transformations.

Consider the situation where we had a merge block that started with an OpPhi, but then we applied a transformation to avoid this OpPhi by storing to and loading from a fresh variable - call this T1.  Then we apply a transformation to add a dead break to the loop.  It won't need any phi ids as the OpPhi is gone - call this T2.

Then during shrinking we try removing transformation T1, so that we try to apply T2 without having applied T1.  In that case we want T2 to turn out to be *inapplicable* because it does not have phi ids, but we do not want the program to abort.

Does that make sense?

Reply:
Makes perfect sense! Basically, we have to make sure that if some two transformations depend on each other (like T2 depends on T1) and if we remove one of them during the shrinking process, the module will remain in a valid state (either with the second transformation or not)

Reply:
Fixed in 2a0b8d1.